### ISSUE TITLE
Scheduling does not take into account attendee content

### ISSUE THREAD

**Issue by Jonty** [emfcamp/Website#1471]
The scheduler does not take attendee-submitted content into account when scheduling things in workshop villages and pretends like it doesn't exist.

Currently if we run the scheduler and take this into account, it will move around attendee content as well as the content we have via the CFP. It will _also_ move any attendee submitted content on a non-content day (e.g. Thursday) to a day when we have content.

There are valid reasons why villages should be able to schedule more than one thing at once, so this isn't necessarily a problem. This issue is to discuss alternatives, and if we should even consider changing this.

The main issues right now:
* Accidentally re-scheduling a workshop over the top of an attendee event without them realising
* Double-booking attendees who are doing an attendee event - this is hard to fix because attendee events are not currently linked to the speaker

**Comment by russss** [comment]
To your first point: once we have venue time ranges (#1150) I think we should have separate time ranges for attendee and non-attendee content. Attendees should not be able to schedule content in a time range reserved for event content. This is the best way to stop any clashing and is more convenient for us - workshop villages can always negotiate this with us.

The preventing-double-booking thing would be nice in theory, but:
* It's quite a lot of hassle for villages to make sure their speaker has an EMF website account and that the proposal is assigned to them.
* It does pose a risk (however slight) of an attendee's action causing the main schedule to become unsolvable.

Ultimately it's not much work for a village to move their own content around, and I'd rather put the onus on them than us. Scheduling one stage is a lot more tractable than scheduling nine.

I'm pretty firmly of the opinion that the scheduler should just ignore attendee content. I don't think any possible benefits outweigh the effort/risks.

**Comment by Jonty** [comment]
Thinking about it, if villages are scheduling content in parallel to the village workshop that is going to be in the _village_ not the village workshop venue, so that's fine. Unsure if we need any real linkage there.

It would be nice to do this on a proposal level, where proposals can (and mostly are) marked as "exclusive" so nothing can be scheduled at the same time as them in any given location. This would give us scheduling flexibility in some of the shared spaces we're starting to consider for drop-in sessions.

I know the workshops team have said that they work with villages to allow them to schedule some things at fixed times (like lunch), but that is just a matter of multiple workshop-type schedulable slots in the village workshop AFAICS.

**Comment by russss** [comment]
> Thinking about it, if villages are scheduling content in parallel to the village workshop that is going to be in the village not the village workshop venue, so that's fine. Unsure if we need any real linkage there.

Yeah, although arguably the proper answer to that is to let villages have multiple venues...

**Comment by Jonty** [comment]
> > Thinking about it, if villages are scheduling content in parallel to the village workshop that is going to be in the village not the village workshop venue, so that's fine. Unsure if we need any real linkage there.
> 
> Yeah, although arguably the proper answer to that is to let villages have multiple venues...

FFFFFFF. Yes, you're right.

The extreme stretch goal here is obviously to let the scheduler schedule village content automatically in their own venues, while automatically avoiding putting things at times when those speakers are on main stages.

I did not expect this conversation to end up with "recursive festival" as the conclusion.

**Comment by Jonty** [comment]
This needs to support a maximum amount of time that is allowed to be used in the time range - e.g. we have an agreed amount of time that we use workshop village tents for (~4h) but that time can be at any time within 10am-6:30pm. Managing this manually at the moment is awful.

**Comment by Jonty** [comment]
We currently have the ability to tell the scheduler to attempt to pack stuff into a specific time range, this probably needs to be taken into account but I have no idea what the best way to do that is when we are breaking out times like this.

**Comment by marksteward** [comment]
I've added a priority column to slot ranges.
